interp/builtin.go

```diff
@@ -668,29 +674,38 @@ func (r *Runner) builtinCode(ctx context.Context, pos syntax.Pos, name string, a
 			}
 		}
 		args := fp.args()
+		bash := !posixOpts
 		if len(args) == 0 {
-			if !posixOpts {
-				for i, name := range bashOptsTable {
-					r.printOptLine(name, r.opts[len(shellOptsTable)+i])
+			if bash {
+				for i, opt := range bashOptsTable {
+					r.printOptLine(opt.name, r.opts[len(shellOptsTable)+i], opt.supported)
 				}
 				break
 			}
 			for i, opt := range &shellOptsTable {
-				r.printOptLine(opt.name, r.opts[i])
+				r.printOptLine(opt.name, r.opts[i], true)
 			}
 			break
 		}
 		for _, arg := range args {
-			opt := r.optByName(arg, !posixOpts)
-			if opt == nil {
-				r.errf("shopt: invalid option name %q\n", arg)
+			v, _, opt := r.optByName(arg, bash)
+			if v == nil {
+				r.errf("bash: line %d: shopt: %s: invalid shell option name\n", pos.Line(), arg)
```

Comment:
I sorta disagree with using `bash:` in error lines, because we are not Bash :)

It's OK if the error lines don't match Bash exactly character by character. See `#JUSTERR` in some of the test cases; those tests check that both shells error, but doesn't require Bash to match the output exactly.